agent/accumulator.go

```diff
@@ -161,6 +164,14 @@ func (ac *accumulator) AddFields(
 	ac.metrics <- m
 }
 
+func (ac *accumulator) AddError(err error) {
+	if err == nil {
+		return
+	}
+	atomic.AddUint64(&ac.errCount, 1)
+	log.Printf("ERROR in input [%s]: %s", ac.inputConfig.Name, err)
```

Comment:
reading ac.inputConfig.Name could produce a race condition here. Since this really shouldn't happen often, I would remove the `atomic`, add a sync.Mutex to the accumulator, and change to:

```
	if err == nil {
		return
	}
	ac.Lock()
	ac.errCount++
	log.Printf("ERROR in input [%s]: %s", ac.inputConfig.Name, err)
	ac.Unlock()
```